src/crawlee/beautifulsoup_crawler/_beautifulsoup_crawling_context.py

```diff
@@ -1,26 +1,21 @@
-from __future__ import annotations
+from dataclasses import dataclass, fields
 
-from dataclasses import dataclass
-from typing import TYPE_CHECKING
+from bs4 import BeautifulSoup
+from typing_extensions import Self
 
-from crawlee._types import BasicCrawlingContext, EnqueueLinksFunction
 from crawlee._utils.docs import docs_group
-from crawlee.http_crawler import HttpCrawlingResult
-
-if TYPE_CHECKING:
-    from bs4 import BeautifulSoup
+from crawlee.abstract_http_crawler._http_crawling_context import ParsedHttpCrawlingContext
 
 
 @dataclass(frozen=True)
 @docs_group('Data structures')
-class BeautifulSoupCrawlingContext(HttpCrawlingResult, BasicCrawlingContext):
-    """The crawling context used by the `BeautifulSoupCrawler`.
-
-    It provides access to key objects as well as utility functions for handling crawling tasks.
-    """
-
-    soup: BeautifulSoup
-    """The `BeautifulSoup` object for the current page."""
-
-    enqueue_links: EnqueueLinksFunction
-    """The BeautifulSoup `EnqueueLinksFunction` implementation."""
+class BeautifulSoupCrawlingContext(ParsedHttpCrawlingContext[BeautifulSoup]):
```

Comment:
Please do not remove the class docstring. Every public should have it. We render it on the API doc website.

Reply:
Done